src/errors.rs

```diff
@@ -27,6 +27,10 @@ pub enum Error {
     StorageUnavailable,
     #[error("The entry acquired has been compacted")]
     StorageCompacted,
+    #[error("Unrecognized log file version: {0}")]
+    UnrecognizedLogFileVersion(u64),
+    #[error("Unrecognized log file magic header")]
+    UnrecognizedLogFileMagicHeader,
```

Comment:
How about `Corruption(#[from] Box<dyn error::Error + Send + Sync>),`? I feel like there are too many errors...

Reply:
And add an unit test for newly introduced error conditions.